bridge/src/monitors/nine-chronicles-transferred-event-monitor.ts

```diff
@@ -1,19 +1,62 @@
-import { ConfirmationMonitor } from "./confirmation-monitor";
+import { TriggerableMonitor } from "./triggerable-monitor";
 import { IHeadlessGraphQLClient } from "../interfaces/headless-graphql-client";
 import { NCGTransferredEvent } from "../types/ncg-transferred-event";
 import { TransactionLocation } from "../types/transaction-location";
 
-export class NineChroniclesTransferredEventMonitor extends ConfirmationMonitor<NCGTransferredEvent> {
+
+const AUTHORIZED_BLOCK_INTERVAL = 50;
+
+export class NineChroniclesTransferredEventMonitor extends TriggerableMonitor<NCGTransferredEvent> {
     private readonly _headlessGraphQLClient: IHeadlessGraphQLClient;
     private readonly _address: string;
 
-    constructor(latestTransactionLocation: TransactionLocation | null, confirmations: number, headlessGraphQLClient: IHeadlessGraphQLClient, address: string) {
-        super(latestTransactionLocation, confirmations);
+    constructor(latestTransactionLocation: TransactionLocation | null, headlessGraphQLClient: IHeadlessGraphQLClient, address: string) {
+        super(latestTransactionLocation);
 
         this._headlessGraphQLClient = headlessGraphQLClient;
         this._address = address;
     }
 
+    protected async processRemains(transactionLocation: TransactionLocation): Promise<{ nextBlockIndex: number; remainedEvents: { blockHash: string; events: any[]; }[]; }> {
```

Comment:
Maybe we can use detailed type in here too.

Reply:
I removed the return types because the types can be inferred from the definition in parent class. Please see 
c249c8d  🙏🏻